docs/Recipes/faking-http-client.md

```diff
@@ -0,0 +1,98 @@
+
+Let's assume that you want to create a fake `HttpClient` so you can dictate the
+behavior of the
+[GetAsync(String)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=net-7.0#system-net-http-httpclient-getasync(system-string))

```

Comment:
I realize the goal isn't to cover every scenario, but it might be helpful to also include an example for `PostAsync` with a request body.

Reply:
I'll work something up.

Reply:
Is this close to what you were hoping for?

```csharp
public async Task PostAsync()
{
    using var response = new HttpResponseMessage
    {
        Content = new StringContent("FakeItEasy is fun")
    };

    var handler = A.Fake<FakeableHttpMessageHandler>();

    using var client = new HttpClient(handler);

    using var postContent = new StringContent("I am posting this.");
    var result = await client.PostAsync("https://fakeiteasy.github.io/docs/", postContent);

    A.CallTo(() => handler.FakeSendAsync(
            A<HttpRequestMessage>.That.Matches(
                m => m.Content!.ReadAsStringAsync().Result == "I am posting this."),
            A<CancellationToken>.Ignored))
        .MustHaveHappened();
}
```

Reply:
Looks good to me! The "sync over async" bit is a little annoying, but I don't think there's much we can do about it... I don't think it can actually cause a deadlock in this case.

Reply:
yes